src/main/scala/io/github/pauljamescleary/petstore/infrastructure/authentication/PasswordHasherCryptInterpreter.scala

```diff
@@ -0,0 +1,17 @@
+package io.github.pauljamescleary.petstore
+package infrastructure.authentication
+
+import cats.effect.Sync
+import tsec.passwordhashers.PasswordHash
+import domain.authentication.CryptAlgebra
+import tsec.passwordhashers.core.PasswordHasher
+
+class PasswordHasherCryptInterpreter[F[_] : Sync, A](
+  implicit H: PasswordHasher[A]
```

Comment:
is there a reason to have the `PasswordHasher` be implicit?

Reply:
Got it, ok

Reply:
Yeah i think this kind of works, because you can just do `new PasswordHasherCryptInterpreter[IO, SCrypt]` to specify which alg you want.